### ISSUE TITLE
The client doesn't obey set-cookie with empty value.

### ISSUE THREAD

**Issue by rulatir** [jeremychone/rust-httpc-test#3]
I am trying to follow [this tutorial](https://www.youtube.com/watch?v=XZtlD_m59sM&t=1481s). I decided to go one tiny inch beyond what is done in the video and implemented a logout route that is supposed to delete the auth cookie. It seems to work OK on the server side, i.e. the server sends a correct empty-valued, expired-last-year set-cookie directive in order to delete the cookie:

```
->> HANDLER      - api_logout
->> RES_MAPPER   - main_response_mapper


=== Response for POST http://localhost:8080/api/logout
=> Status         : 200 OK
=> Headers        :
   content-type: "application/json"
   content-length: "4"
   set-cookie: "auth-token=; Max-Age=0; Expires=Tue, 24 May 2022 23:05:08 GMT"
   date: "Wed, 24 May 2023 23:05:08 GMT"
=> Response Cookies:
   auth-token: 
=> Client Cookies :
   auth-token: user-1.exp.sign
=> Response Body  :
"Ok"
===
```

However subsequent requests made with the same client keep sending the old cookie, as if the client didn't apply the directive:

```
->> HANDLER      - handler_hello2 - "Rulatir"
->> RES_MAPPER   - main_response_mapper


=== Response for GET http://localhost:8080/hello/Rulatir
=> Status         : 200 OK
=> Headers        :
   content-type: "text/html; charset=utf-8"
   content-length: "30"
   date: "Wed, 24 May 2023 23:05:08 GMT"
=> Client Cookies :
   auth-token: user-1.exp.sign
=> Response Body  :
Hello <strong>Rulatir</strong>
===
```

**Comment by jeremychone** [comment]
Hum, I will look at it, but I am not sure this an `httpc-test` issue. 

The httpc-test uses `reqwest` and the `reqwest client store` and it would be surprising if those libs would have this bug. 

I will eventually implement the logout in the `rust-web-app` awesomeapp blueprint (video coming soon), and report back here. 

Thanks for reporting though.

**Comment by rulatir** [comment]
Oh, you're the author of that tutorial :)

**Comment by SFToro** [comment]
Arrived to the exact same point trying to do that while following the video tutorial. 
Same behaviour (works ok but prints wrong info).

**Comment by jeremychone** [comment]
Ha, the prints are the issue. I might have a bug there. Now I know where to look. 

Anyway, finishing the video, and did not do the logoff yet. But will do soon.

**Comment by jeremychone** [comment]
ok, sorry @SFToro @rulatir I think I made a silly mistake in `httpc-test` about the cookies. 

continuing to investigate and will fix soon.

**Comment by jeremychone** [comment]
Ok, wrong alert, `httpc-test` was fine. I thought it was a `drain` I did on the `HeaderMap`, but in fact, it was ok and the right thing to do (to avoid clones). 

Now, I found the issue. In the `Rust Axum Full Course` I do not set the cookie path to root, and so, the cookies get rightfully set to only `/api/`. 

So, the fix is to do the `set_path` as follow. 

```rs
	let mut cookie = Cookie::new(AUTH_TOKEN, token.to_string());
	cookie.set_http_only(true);
	cookie.set_path("/");

	cookies.add(cookie);
	...
	```

And then it should work. 

I am closing this issue as I think this was the issue you both have. If not, feel free to log another ticket with some simple code sample I can reproduce.
